s3 integration tests

s-hertel opened the pull request:
##### SUMMARY
Moving integration tests originally in #21529 so it's easier to verify that they work for the current s3 and that boto3ified s3 meets expectations. 

##### ISSUE TYPE
 - Test Pull Request

##### COMPONENT NAME
test/integration/amazon.yml
test/integration/roles/test_s3/tasks/main.yml

##### ANSIBLE VERSION
```
ansible 2.4.0 (s3-integration-tests ddf85acc7d) last updated 2017/03/23 16:19:04 (GMT -400)
  config file =
  configured module search path = Default w/o overrides
```

s-hertel committed:
s3 integration tests

mikedlr committed:
Make s3 integration tests work in standard integration setup and with fish as current shell
see also https://github.com/ansible/proposals/issues/62 if you don't like this much

mikedlr committed:
test_s3 integration test - substitute command instead of shell where simple

mikedlr committed:
test_s3 integration test - random contents + complete elimination of use of shell module by moving to tempfile

s-hertel committed:
move s3 to integration test targets

ryansb commented on test/integration/targets/s3/aliases:
```diff
@@ -0,0 +1,2 @@
+cloud/aws
+posix/ci/cloud/aws
```
I'm not familiar with what each of these aliases is for, but is S3 really something that counts as POSIX? Or is that just to denote the OS type?

s-hertel replied:
Oh. Good catch, thanks.

s-hertel committed:
Add a couple more test for . in bucket names

Tidy syntax

remove reference to legacy credentials.yml

remove posix/cli/cloud/aws

s-hertel committed:
remove from legacy testing

s-hertel committed:
remove dependencies that aren't used
